modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -277,7 +277,18 @@ private TransactionConfig buildTransactionConfig() {
     MuleTransactionConfig transactionConfig = new MuleTransactionConfig();
     transactionConfig.setAction(toActionCode(sourceAdapter.getTransactionalAction()));
     transactionConfig.setMuleContext(muleContext);
-    transactionConfig.setFactory(transactionFactory);
+
+    TransactionType transactionalType = sourceAdapter.getTransactionalType();
+    switch (transactionalType) {
```

Comment:
why is this different? The lookup logic should be able to pick both kinds of transaction factories